### ISSUE TITLE
Language filtering is broken

### ISSUE THREAD

**Issue by tzdybal** [dgraph-io/dgraph#1118]
Simple example.
Schema:
```
mutation { schema { name: string @index(term) .} }
```
Data:
```
mutation { 
  set { <0x01> <name> "Sam" . } 
  set { <0x01> <name> "Bob"@en . }
}
```
Query:
```
query { me(func:eq(name@en, "Bob")) { name name@en _uid_ } }
```
The query above gives no results.

Everything works fine when `exact` index is used.

**Comment by tzdybal** [comment]
There are two separate errors. One is related to filtering itself. Second is a problem w mutation handling - see #1119.

**Comment by tzdybal** [comment]
Merged to master (a5bcb620c03da496caf51db57e1afbba79d9ab9d).
